Add BIRD2 language

Alice39s opened the pull request:
## Summary
  
This PR introduces syntax highlighting support for the **BIRD2 configuration language** via `.conf` files used by the [BIRD Internet Routing Daemon](https://bird.network.cz/). 

Unlike simpler configuration models (e.g., INI or Nginx), BIRD employs a uniquely intricate paradigm with distinct implementation challenges. We address this by adding a `textmate` grammar with precise scoping for:

- Protocol definitions  
- Filter functions  
- Route maps  
- Other core BIRD2 constructs  

## Motivation

BIRD2 powers critical Internet infrastructure at scale, including:  
- **AMS-IX** – 870+ ASNs, 14Tb/s traffic ([ref](https://www.ams-ix.net/ams/documentation/ams-ix-route-servers))  
- **LINX** – 1,000+ BGP sessions on v2.13 ([ref](https://www.linx.net/wp-content/uploads/2024/05/Day-1-P4-LINX_Technology-Presentation_v3.0.pdf))  
- **Cloudflare** – Anycast routing across all edge PoPs ([ref](https://blog.cloudflare.com/cloudflares-architecture-eliminating-single-p/))  

Despite its critical role in major Internet infrastructure, editors like VSCode and Shiki lack native grammar support for BIRD2 configurations. This PR aims to fill that gap.

## References

- **Grammar Source**: [bird-chinese-community/BIRD-tm-language-grammar](https://github.com/bird-chinese-community/BIRD-tm-language-grammar)  
- **Main Contributors**: [Alice39s](https://github.com/Alice39s), [pppwaw](https://github.com/pppwaw)

## License

- Syntax files are distributed under **[Mozilla Public License 2.0](https://github.com/bird-chinese-community/BIRD-tm-language-grammar/blob/main/LICENSE.syntax)**
- Sample configuration files are distributed under **[MIT License](https://github.com/bird-chinese-community/BIRD-tm-language-grammar/blob/main/LICENSE.sample)**

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      - [Public code rearch results](https://github.com/search?q=%22protocol+bgp%22+OR+%22neighbor%22+OR+%22local+as%22+path%3A*.conf+NOT+is%3Afork&type=code&ref=advsearch) -- **27k+** BIRD2 configuration snippets found in public repositories
      - [Public repo rearch results](https://github.com/search?q=bird+config&type=repositories&ref=advsearch) -- **883+** active repositories using BIRD configurations
      - `bird.conf`: 1.3k+ https://github.com/search?q=path%3Abird.conf&type=code
      - `bird6.conf`: 275+ https://github.com/search?q=path%3Abird6.conf&type=code
      - `bird2.conf`: 149+ https://github.com/search?q=path%3Abird2.conf&type=code

  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source:
      - https://github.com/bird-chinese-community/BIRD-tm-language-grammar/blob/main/sample/
    - Sample license: **[MIT License](https://github.com/bird-chinese-community/BIRD-tm-language-grammar/blob/main/LICENSE.sample)**
  - [x] I have included a syntax highlighting grammar: https://github.com/bird-chinese-community/BIRD-tm-language-grammar/blob/main/Syntaxes/bird2.tmLanguage.json
  - [x] I have added a color
    - Hex value: `#b6d7e4`
    - Rationale: This color comes from the [Official Documentation](https://bird.network.cz/) logo and accent colors.
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

Alice39s committed:
Add BIRD2 language

Alice39s committed:
Add BIRD2 language support

lildude reviewed (changes requested):
See inline comment. Please also follow the instructions for adding the grammar in CONTRIBUTING.md.

Alice39s committed:
refactor(linguist): refine BIRD2 language detection by removing .conf heuristic and adding dedicated extensions

Alice39s committed:
Add `BIRD2` syntax highlighting grammar and fix alias conflict

Alice39s committed:
Add BIRD2 syntax highlighting grammar and fix alias conflict

lildude reviewed (changes requested):
Don't remove the two missing grammar licences. I'm trying to get the author to restore the repos.

Alice39s commented:
> Don't remove the two missing grammar licences. I'm trying to get the author to restore the repos.

You're right — thanks for pointing that out.

I ran `script/add-grammar` and didn't realize it would remove those entries automatically. That wasn’t my intention, and I’ll make sure to fix it. Apologies for the oversight.

Alice39s commented on lib/linguist/languages.yml:
```diff
@@ -634,6 +634,29 @@ Bikeshed:
+  - ".bird.conf"
+  - ".bird2.conf"
+  - ".bird3.conf"
+  - ".birdconf"
```
Thanks for the review!

You're right — `.bird.conf` is in use today (e.g. [[examples here](https://github.com/search?q=.bird.conf&type=code)](https://github.com/search?q=.bird.conf&type=code)).

The others (`.bird2.conf`, `.bird3.conf`, `.birdconf`) were added in anticipation of future usage, but I understand Linguist's policy on current evidence.

I'm happy to drop them if needed and revisit later as adoption grows.

Alice39s committed:
chore: restore removed grammar licenses for sublime-nginx and sublime-varnish

Alice39s committed:
chore: remove low-usage extensions (.bird2.conf, .bird3.conf, .birdconf)

Alice39s committed:
fix: remove trailing blank line accidentally added

Alice39s commented:
Hi @lildude  👋

Gentle ping on this — updates done per your feedback (extensions & licenses). Could you take a look when you have a moment?

Thanks for your time!

lildude reviewed (changes requested):
See inline comments. Additionally, as you're adding support only for filenames, the samples need to match those filenames. None of your samples do so they need to be replaced with real examples of those you're adding support for.

Alice39s committed:
Fix incorrect license

Co-authored-by: Colin Seymour <colin@github.com>

Alice39s committed:
fix: remove unused extensions and adjusting filenames

Alice39s committed:
chore: add new advanced BGP configuration

Alice39s committed:
feat: add basic BIRD2 configuration with routing protocols and filters

Alice39s reviewed (commented):
Resolved https://github.com/github-linguist/linguist/pull/7513#pullrequestreview-3110504561

@lildude

lildude reviewed (changes requested):
Samples can not be in subdirectories. Please move them to a level.

Please also update the PR template to provide individual links to the search results for each of the filenames you're adding and fix the information about the samples' sources and licences.

Alice39s committed:
fix: Adjust the structure of BIRD2 sample files

- Move all BIRD2 sample configuration files from subdirectories to the top-level directory ().

Alice39s commented:
> Samples can not be in subdirectories. Please move them to a level.
> 
> Please also update the PR template to provide individual links to the search results for each of the filenames you're adding and fix the information about the samples' sources and licences.

@lildude Thanks for the feedback and detailed explanation!

I've made the following changes based on your suggestions:

1. I've moved all the sample configuration files from subdirectories to the top level. They are now all located in the `samples/BIRD2` directory.
2. I've updated the PR template to provide individual search result links for each new filename.
3. I've corrected the information regarding the samples' sources and licenses.

Please take another look, and let me know if there are any further changes required.

Alice39s committed:
fix: remove unused BIRD2 configuration files and clean up language definitions

Alice39s reviewed (commented):
Resolved https://github.com/github-linguist/linguist/pull/7513#discussion_r2281682779

@lildude

Alice39s committed:
Fix incorrect sorting

Alice39s committed:
Fix incorrect sorting

Alice39s committed:
Merge branch 'main' into add-bird2-language

Alice39s commented:
Hi @lildude 👋

Hope you’re doing well! Just wanted to gently follow up on this PR when you have a moment. I believe all the requested changes have been addressed, and it’s currently awaiting your review.

Please let me know if there’s anything else you’d like me to adjust. Really appreciate your time and guidance on this!

Thanks in advance 🙏

Alice39s committed:
Merge branch 'main' into add-bird2-language

Alice39s committed:
Merge branch 'main' into add-bird2-language

lildude reviewed (changes requested):
The test failure is because `samples/BIRD2/bird.conf` is in the wrong place. It needs to be at `samples/BIRD2/filenames/bird.conf`

Alice39s committed:
fix: move sample BIRD2 config to correct path

Alice39s commented:
> The test failure is because `samples/BIRD2/bird.conf` is in the wrong place. It needs to be at `samples/BIRD2/filenames/bird.conf`

Thanks for catching that! 🙏 I've moved the sample config to `samples/BIRD2/filenames/bird.conf` as suggested.  

Please let me know if there’s anything else you'd like me to adjust.

Alice39s committed:
Merge branch 'main' into add-bird2-language

Alice39s committed:
Merge branch 'main' into add-bird2-language

lildude committed:
Merge branch 'main' into add-bird2-language
